bool lift
bool drop
bool save_weight